### PR TITLE
backend:removing unencrypted number

### PR DESCRIPTION
**Pull request by vijaygupta18**
## Type of Change

- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement
- [x] Refactoring
- [ ] Dependency updates

### Additional Changes

- [ ] This PR modifies the database schema (database migration added)
- [ ] This PR modifies dhall configs/environment variables
- [ ] This PR contains API breaking changes

## How did you test it?
image

## Checklist

- [ ] I formatted the code and addressed linter errors `./dev/format-all-files.sh`
- [ ] I reviewed submitted code
- [ ] I added unit tests for my changes where possible
- [ ] I added integration tests for my changes where possible
- [ ] I tested the changes using integration tests
- [ ] I added a [CHANGELOG](/CHANGELOG.md) entry if applicable
- [ ] No leak detected in leakcanary

### CODE REVIEW

File: Backend/app/rider-platform/rider-app/Main/spec/Storage/ride.yaml [hunk 1]
```diff
@@ -20,51 +21,16 @@ imports:
...
 Ride:
   tableName: ride
-  derives: "Generic,Show"
+  derives: "Generic"
   types:
     RideStatus:
       enum: "NEW,INPROGRESS,COMPLETED,CANCELLED"
       derive: "HttpInstance"
     BPPRide: {}
-    RideAPIEntity:
```
Comment (hkmangla) [inline Backend/app/rider-platform/rider-app/Main/spec/Storage/ride.yaml:32]: Why did you remove this?

**Review by hkmangla — DISMISSED** [review]

File: Backend/app/rider-platform/rider-app/Main/spec/Storage/ride.yaml (hunk 1 above)
Comment (vijaygupta18) [reply Backend/app/rider-platform/rider-app/Main/spec/Storage/ride.yaml:32]: api entity shouldn't be in storage file..aslo keeping it here will leads to ambiguity in the beamType

File: Backend/app/rider-platform/rider-app/Main/src/Domain/Action/Beckn/Common.hs [hunk 2]
```diff
@@ -442,7 +443,8 @@ rideCompletedReqHandler ValidatedRideCompletedReq {..} = do
   fork "updating total rides count" $ SMC.updateTotalRidesCounters booking.riderId
   merchantConfigs <- CMC.findAllByMerchantOperatingCityId booking.merchantOperatingCityId
   SMC.updateTotalRidesInWindowCounters booking.riderId merchantConfigs
-  mbAdvRide <- QRide.findLatestByDriverPhoneNumber ride.driverMobileNumber
+  driverPhoneNumber <- mapM decrypt ride.driverPhoneNumber >>= fromMaybeM (RideFieldNotPresent "driverPhoneNumber")
+  mbAdvRide <- QRide.findLatestByDriverPhoneNumber driverPhoneNumber
```
Comment (piyushKumar-1) [inline Backend/app/rider-platform/rider-app/Main/src/Domain/Action/Beckn/Common.hs:447]: lookup using hash here, and fallback to number based if not found.

**Review by piyushKumar-1 — APPROVED** [review]
